administrator/components/com_associations/helpers/associations.php

```diff
@@ -89,8 +90,22 @@ public static function getItemTypeProperties($key = '')
 			JLoader::register($componentName . 'Model' . $itemName, $modelsPath . '/' . $it[$key]->item . '.php');
 			$it[$key]->model = JModelLegacy::getInstance($itemName, $componentName . 'Model', array('ignore_request' => true));
 
-			// If component item type model cannot loaded, or associations properties does not exist, item type does not support associations.
-			$it[$key]->associations->support = $it[$key]->model && $it[$key]->model->get('associationsContext');
+			// Load component's helper
+			$helperName = $componentName . 'AssociationsHelper';
+			JLoader::register($helperName, $helpersPath . '/associations.php');
+
+			// If component item helper cannot loaded, or hasAssociationsSupport is false, item type does not support associations.
+			if (file_exists($helpersPath . '/associations.php') && $it[$key]->item !== 'category')
```

Comment:
you register it with JLoader before and only after you check if the file exists? doen't make a lot of sense to me, please check.